.github/workflows/release-docker.yml

```diff
@@ -0,0 +1,34 @@
+---
+name: Release Docker images
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  plugin-compiler:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout of tyk
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - uses: docker/setup-buildx-action@v2
+
+      - id: docker-tag
+        uses: yuya-takeyama/docker-tag-from-github-ref-action@v1
+
+      - name: Plugin compiler
+        uses: docker/build-push-action@v3
+        with:
+          push: false         # startsWith(github.ref, 'refs/tags')
```

Comment:
How are you planning to push this image?

Reply:
1. remove plugin compiler from goreleaser,
2. always push

The docker tag select would fall on `latest` for master/main, but the issue is we use `release-5.1.0` rather than `v5.1.0`, but we also tag `v5.1.0` so I'm not sure if we need to push:true or just push on tag. It still builds the images without a push if nothing is tagged, so I am happy with that.

https://github.com/marketplace/actions/docker-tag-from-github-ref